### PR TITLE
THRIFT-4949 improve HTTP/1 server test case

### PR DESCRIPTION
**Pull request by Issacpeng**
improve HTTP/1 server test case

  

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?  (not required for trivial changes)
- [ ] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [ ] Did you squash your changes to a single commit?  (not required, but preferred)
- [ ] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [ ] If your change does not involve any code, add ` [skip ci]` at the end of your pull request to free up build resources.

### CODE REVIEW

File: lib/java/test/org/apache/thrift/test/TestClient.java [hunk 1]
```diff
@@ -145,8 +149,15 @@ public static void main(String [] args) {
...
-        String url = "http://" + host + ":" + port + "/service";
-        transport = new THttpClient(url);
+        String url = "http://" + host + ":" + port + "/test/service";
+        if (http_client == true) {
+          
+          transport = new THttpClient(url, HttpClients.createDefault());
+        } else {
+          transport = new THttpClient(url);
+        }
+
```
Comment (qinghui-xu) [inline lib/java/test/org/apache/thrift/test/TestClient.java]: Remove the empty lines, please.

File: lib/java/test/org/apache/thrift/test/TestServlet.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      return tProtocolFactory;
+  }
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  @Override
+  protected TProcessor getProcessor(){
+      TestHandler testHandler = new TestHandler();
+      ThriftTest.Processor testProcessor = new ThriftTest.Processor(testHandler);
+      return testProcessor;
+
```
[the remark this answers is not in the document]
Comment (Issacpeng) [reply lib/java/test/org/apache/thrift/test/TestServlet.java]: update for that

File: lib/java/test/org/apache/thrift/test/TestClient.java (hunk 1 above)
Comment (Issacpeng) [reply lib/java/test/org/apache/thrift/test/TestClient.java]: update for that

File: lib/java/gradle.properties [hunk 3]
```diff
@@ -28,7 +28,9 @@ maven-repository-id=apache.releases.https
 httpclient.version=4.5.6
 httpcore.version=4.4.1
 slf4j.version=1.7.25
-servlet.version=2.5
+#servlet.version=2.5
+#It contains servlet3
+tomcat.embed.version=8.5.5
```
Comment (qinghui-xu) [inline lib/java/gradle.properties]: Also, is there a reason not to use tomcat 9.x?

Comment (Issacpeng) [reply lib/java/gradle.properties]: Apache Tomcat 8.5 is one of the most frequently used Tomcat versions, and we tend to choose the most popular and stable version.

Comment (qinghui-xu) [reply lib/java/gradle.properties]: If we decide to pick 8.5, maybe it's better to pick the latest one (8.5.46).

Comment (Issacpeng) [reply lib/java/gradle.properties]: checked it and agree with you, down that.

**Comment by dcelasun** [comment]
Tests are failing:

```
> Task :compileCrossTestJava FAILED
/thrift/src/lib/java/test/org/apache/thrift/test/TestClient.java:80: error: cannot find symbol
    boolean http_client = fasle;
                          ^
  symbol:   variable fasle
  location: class TestClient
```

**Comment by Issacpeng** [comment]
@dcelasun  updated to fix that problem

File: lib/java/gradle/environment.gradle [hunk 4]
```diff
@@ -66,7 +67,8 @@ dependencies {
     compile "org.slf4j:slf4j-api:${slf4jVersion}"
     compile "org.apache.httpcomponents:httpclient:${httpclientVersion}"
     compile "org.apache.httpcomponents:httpcore:${httpcoreVersion}"
-    compile "javax.servlet:servlet-api:${servletVersion}"
+    //compile "javax.servlet:servlet-api:${servletVersion}"
+    compile "org.apache.tomcat.embed:tomcat-embed-core:${tomcatEmbedVersion}"
```
Comment (afs) [inline lib/java/gradle/environment.gradle:71]: Was "testCompile" meant here? As is, this becomes a dependency downstream.

[THRIFT-5375](https://issues.apache.org/jira/browse/THRIFT-5375).

**Pull request merged**